pkg/pagerduty/pagerduty.go

```diff
@@ -126,6 +128,76 @@ func (p PagerDuty) AddNote(incidentID string, noteContent string) error {
 	return nil
 }
 
+// ExtractIDFromCHGM extracts from an Alert body until an external ID
+// the function is a member function but doesn't use any of the other funcs / types in the PagerDuty struct
+func (_ PagerDuty) ExtractIDFromCHGM(data map[string]interface{}) (string, error) {
+	var err error
+
+	externalBody, err := extractNotesFromBody(data)
+	if err != nil {
+		return "", fmt.Errorf("cannot marshal externalCHGMAlertBody: %w", err)
+	}
+
+	if externalBody == "" {
+		return "", AlertBodyDoesNotHaveNotesFieldErr{}
+	}
+
+	internalBody := internalCHGMAlertBody{}
+	err = yaml.Unmarshal([]byte(externalBody), &internalBody)
+	if err != nil {
+		return "", NotesParseErr{Err: err}
+	}
+
+	if internalBody.ClusterID == "" {
+		return "", AlertNotesDoesNotHaveClusterIDFieldErr{}
+	}
+
+	return internalBody.ClusterID, nil
+}
+
+// extractNotesFromBody will extract from map[string]interface{} the '.details.notes' while doing type checks
+// this is better than a third party as it is better maintained and required less dependencies
+func extractNotesFromBody(body map[string]interface{}) (string, error) {
+	var ok bool
+	_, ok = body["details"]
+	if !ok {
+		return "", AlertBodyExternalParseErr{FailedProperty: ".details"}
+	}
+
+	details, ok := body["details"].(map[string]interface{})
+	if !ok {
+		err := AlertBodyExternalCastErr{
+			FailedProperty: ".details",
```

Comment:
What's about adding the body here as a parameter? When you print the error, you immediately got the input that has caused this exception

Reply:
done